pkg/policies/policies.go

```diff
@@ -183,6 +195,81 @@ func (pv *PolicyVerifier) evaluatePolicyAttachment(ctx context.Context, attachme
 	}, nil
 }
 
+func (pv *PolicyVerifier) computeArguments(inputs []*v1.PolicyInput, args map[string]string, bindings map[string]string) (map[string]string, error) {
+	result := make(map[string]string)
+
+	// Policies without inputs in the spec
+	// TODO: Remove this in next release, once users have migrated their policies
+	if len(inputs) == 0 {
+		result = args
+	}
+
+	// Check for required inputs
+	for _, input := range inputs {
+		// Illegal combination
+		if input.Required && input.Default != "" {
+			return nil, fmt.Errorf("required input %s with a default value is illegal", input.Name)
+		}
+		if _, ok := args[input.Name]; !ok {
+			if input.Required {
+				return nil, fmt.Errorf("missing required input %q", input.Name)
+			}
+			// if not required, and it has a default value, let's use it
+			if args[input.Name] == "" && input.Default != "" {
+				value, err := applyBinding(input.Default, bindings)
+				if err != nil {
+					return nil, err
+				}
+				result[input.Name] = value
+			}
+		}
+	}
+
+	// check for provided arguments
+	for k, v := range args {
+		expected := slices.ContainsFunc(inputs, func(input *v1.PolicyInput) bool {
+			return input.Name == k
+		})
+		if !expected {
+			pv.logger.Warn().Msgf("argument %q will be ignored", k)
+			continue
+		}
+		value, err := applyBinding(v, bindings)
+		if err != nil {
+			return nil, err
+		}
+		result[k] = value
+	}
+
+	return result, nil
+}
+
+// renders an input template using bindings in Go templating format
+func applyBinding(input string, bindings map[string]string) (string, error) {
+	if bindings == nil {
+		return input, nil
+	}
+
+	// Support both `.inputs.foo` and `inputs.foo`
+	input = inputsPrefixRegexp.ReplaceAllString(input, "{{ .inputs.")
```

Comment:
what if there is no space? `{{.inputs...)` can't it be done with an actual regex 

/$\.inputs/

Reply:
That's not the regex, that's the final replacement. The regex looks like this (check top of the file) `{{\s*(inputs.)

Reply:
re: what if there is no space?. All cases are tested, no spaces, multiple spaces, ...